src/transformers/models/longt5/modeling_longt5.py

```diff
@@ -404,11 +419,14 @@ def _relative_position_bucket(relative_position, bidirectional=True, num_buckets
         relative_buckets += torch.where(is_small, relative_position, relative_position_if_large)
         return relative_buckets
 
-    def compute_bias(self, query_length, key_length, device=None):
+    def compute_bias(self, query_length, key_length, device=None, cache_position=None):
```

Comment:
This is the best I could come up with to overcome dynamic control. Otherwise the `real_seq_length` is always depedent on cache length which is currently computed from tensor values being not 0

Basically this is similar as if we used position ids, but T5 models don't pass position ids. Cache position is exactly what we need then

Reply:
sounds good!